src/web/js/google-apis/picker.js

```diff
@@ -0,0 +1,128 @@
+
+  function onPickerLoaded(drive) {
+    return function() {
+      this.init(drive, google.picker);
+      this.onLoaded();
+    };
+  }
+
+  storageAPI.then((function(drive){
+    google.load('picker', '1', { 'callback': onPickerLoaded(drive).bind(this) });
+  }).bind(this));
+}
+
+FilePicker.prototype.init = function(drive, picker) {
+  this.open = this.initOpen(drive, picker);
+  console.log("Initializing File Picker");
+  // this.openOn does not need initialization
+};
+
+FilePicker.prototype.initOpen = function(drive, picker) {
+
+  var validateSelection = (function(sel) {
+    if (sel.length === 0) {
+      // Should be impossible, since we only run this if the user
+      // has picked something (see pickerCallback)
+      this.onInternalError("Internal Error (please report to developers): "
+                           + "Should be impossible: "
+                           + "Picker selection length is zero");
+    } else {
+      var firstType = sel[0][picker.Document.TYPE];
+      // Check that all selected files are of the same type
+      sel.forEach((function(f) {
+        if (f[picker.Document.TYPE] !== firstType) {
+          this.onError("Cannot open files of multiple types "
+                       + "(Expected: " + firstType
+                       + "; Received: " + f[picker.Document.TYPE]
+                       + ")");
+        }
+      }).bind(this));
+      switch (sel[0][picker.Document.TYPE]) {
+      case "file": // Pyret file
+        break;
+      case picker.Type.PHOTO: // Photo
+        // Fix photo permissions
+        var permissions = {
+          'role': 'reader',
+          'type': 'anyone',
+          'value': 'default',
+          'withLink': true
+        };
+        sel.forEach((function(photo) {
+          var fileId = photo[picker.Document.ID];
+          gwrap.drive.permissions.insert({
+            'fileId': fileId,
+            'resource': permissions
+          }).catch(this.onInternalError);
+        }).bind(this));
+        break;
+      default:
+        this.onError("Cannot open file of unknown type: " + sel[0][picker.Document.TYPE]);
+      }
+    }
+  }).bind(this);
+
+  var pickerCallback = (function(data) {
+    if (data[picker.Response.ACTION] == picker.Action.PICKED) {
+      var selection = data[picker.Response.DOCUMENTS];
+      validateSelection(selection);
+      this.dataHandler(selection, picker, drive);
+    }
+  }).bind(this);
+
+  var showPicker = (function() {
+    /**
+     * A Picker View which displays Pyret files which users may load.
+     */
+    var pyretView = new picker.DocsView(picker.ViewId.DOCUMENTS);
+
+    // [Image picker code is based on the Picker code from WeScheme:
+    // https://github.com/bootstrapworld/wescheme2012/blob/5b03bac247ac182c356d73fc3e7283ef1086777f/war-src/js/openEditor/editor.js#L571-L659 ]
+    /**
+     * A Picker View which displays images users may load.
+     */
+    var imageView = new picker.View(picker.ViewId.DOCS);
```

Comment:
@jpolitz and I were discussing this -- currently, Pyret files are stored with the `text/plain` MIME type (and, yes, now that I think about it, we should probably limit the view to that type), so there is no MIME-filtering way to only show Pyret files (we do, however, start the Pyret file view in the Pyret save folder, which should usually only contain Pyret files). The implications of a Pyret-specific MIME type need to be looked into, but until then, this will have to do.

@schanzer, your comment sounds like it is in reference to images specifically. A few lines down, I have the MIME-filtering enabled for image files. Thus, there shouldn't be any way to open non-images as images (unless one is deliberately trying).